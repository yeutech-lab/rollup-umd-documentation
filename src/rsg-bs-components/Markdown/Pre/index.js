export { default } from 'rsg-components/Markdown/Pre/PreRenderer';
