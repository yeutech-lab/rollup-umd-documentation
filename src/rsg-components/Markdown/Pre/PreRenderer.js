import React from 'react';
import PropTypes from 'prop-types';

import Styled from 'rsg-components/Styled';

const styles = ({ space, color, fontSize, fontFamily, borderRadius }) => ({
	pre: {
		fontFamily: fontFamily.base,
		fontSize: fontSize.small,
		lineHeight: 1.5,
		color: color.base,
		whiteSpace: 'pre',
		backgroundColor: color.codeBackground,
		padding: [[space[1], space[2]]],
		border: [[1, color.border, 'solid']],
		borderRadius,
		marginTop: 0,
		marginBottom: space[2],
	},
});

export function PreRenderer({ classes, children }) {
	return <pre className={classes.pre}>{children}</pre>;
}
PreRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node.isRequired,
};

export default Styled(styles)(PreRenderer);
