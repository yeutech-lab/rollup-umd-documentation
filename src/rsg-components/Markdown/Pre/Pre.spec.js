import React from 'react';

import Pre from './index';

describe('Markdown Pre', () => {
	it('should render a pre', () => {
		const actual = render(<Pre>This is pre-formatted text.</Pre>);

		expect(actual).toMatchSnapshot();
	});
});
