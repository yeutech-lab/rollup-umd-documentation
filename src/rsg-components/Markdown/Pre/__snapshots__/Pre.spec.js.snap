// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Pre should render a pre 1`] = `
<pre
  class="rsg--pre-0"
>
  This is pre-formatted text.
</pre>
`;
